### DISCUSSION TITLE
Leader of the reth repo please open new good issue to work on it

### DISCUSSION THREAD

**Discussion by DoTheBestToGetTheBest** [paradigmxyz/reth#4896] [Help] [answered]
I'am really really interested to work in issue, if by any chance someone can open good issue to work on it this day, i take them all 

I'am really wanna to learn more about reth and ethereum.

I also would give a big thanks to mattsse, he helped me so much

If there is something i can work on it please let me know

I'am excited to work on this thank you !

**Comment by mattsse** [comment, marked as answer]
can't think of anything specific rn, besides the existing good first issues.
will let you know if there's something new

**Reply by DoTheBestToGetTheBest** [reply]
Thank you so much boss !

**Comment by DoTheBestToGetTheBest** [comment]
Good morning @mattsse .  I hope you're doing good. Anything to work on it today please ? :)

**Comment by DoTheBestToGetTheBest** [comment]
@mattsse hey Mat, i hope everything is doing good for you, any issue i can take please ? :)

**Comment by DoTheBestToGetTheBest** [comment]
hey @mattsse please anything to take ? if it possible i want to be assigned as possible, to work on reth issue ! Thank you!

**Comment by DoTheBestToGetTheBest** [comment]
hey @mattsse would love to take others thing if possible :)

**Comment by DoTheBestToGetTheBest** [comment]
@mattsse hey matt, hope you're good ! please assign me directly in issue without asking if i take it or not. Please. Thank you @mattsse

**Comment by mattsse** [comment]
I'll think of something

**Comment by DoTheBestToGetTheBest** [comment]
> I'll think of something

Please don't hesitate to assign on many thing. I really want to do lot of work in reth :) i'm just waiting your assignement. Thank you so much Mat !

**Comment by DoTheBestToGetTheBest** [comment]
@mattsse please don't hesitate to also assign me in other repository if you have something, like foundry, or anything that you judge needed. Thank you. I appreciate everything. Thank you so much

**Comment by DoTheBestToGetTheBest** [comment]
Yo @mattsse please assign me on some new stuff, currently i'm having  homework etc so would i would work on it directly when i finish. Thank you to much Matt

**Comment by mattsse** [comment]
got it, will think of something and tag you

**Reply by DoTheBestToGetTheBest** [reply]
Please don't tag me, directly assign me

**Reply by mattsse** [reply]
opened two issues, 
also feel free to dm on telegram: @mattsse

**Reply by DoTheBestToGetTheBest** [reply]
Thank you so so much @mattsse You are amazing, i'm also very sorry if i'm asking from you too much thing.

**Reply by mattsse** [reply]
looks like you got sniped again -.-

I'll think about smth else

for example we could use some builder style function for this type:

https://github.com/paradigmxyz/reth/blob/05044c189b3473fd8f2fbb2ba7fadaeab467f040/crates/rpc/rpc-types/src/eth/transaction/request.rs#L45-L45

similar to all the functions here that set fields:
https://github.com/paradigmxyz/reth/blob/05044c189b3473fd8f2fbb2ba7fadaeab467f040/crates/transaction-pool/src/test_utils/gen.rs#L113-L113

**Reply by DoTheBestToGetTheBest** [reply]
Please @mattsse make sure that i'am assigned in the issue, in previous issue i wasn't assigned. Thank you !

**Reply by mattsse** [reply]
unfortunately, github only allows this once someone has interacted with the issue...

**Reply by DoTheBestToGetTheBest** [reply]
Alright, could i get 3 issue for me ? I'm thinking taking one good first issue, one medium and one more advanced. Let me know if you are okey with this Mat, thank you for your help. I will comment in the issue now to get the role assigned

**Reply by mattsse** [reply]
great!

do you want to create one for: 
https://github.com/paradigmxyz/reth/discussions/4896#discussioncomment-7320444

I'll think about something medium in the morning

also if you have telegram we can move the convo there

**Reply by DoTheBestToGetTheBest** [reply]
Don't ask me if i want, everything that you have mind i take it. All thing you think about it, i take them and ready to get assigned for it. Will work on it just after homework etc 

Thank you so much Mattsse

I'm really sorry i don't have telegram now, will create one soon !

**Comment by DoTheBestToGetTheBest** [comment]
@mattsse hey Mat, hope you're doing good. Anything new to take ? :)

**Reply by mattsse** [reply]
I left some comments on your recent pr,
do you have keybase,twitter? anything where we can DM, communicating via gh discussions has it's limitations hehe.

I have something in mind which is a bit more advanced, but I think you can take it on.

we need more support for rpc testing. we have these extension helpers for example:

https://github.com/paradigmxyz/reth/blob/9dd270e4e8095a4cab1d2118f8f965eec68f4db7/crates/rpc/rpc-testing-util/src/trace.rs#L23-L37

but we also want more, like a stream for `replayTransaction` that takes a list of hashes and returns the traced results in a stream.
this requires some additional knowledge about futures/streams etc, but I think you can take this now.
happy to give access to a node so you can test this properly.

wdyt?

**Reply by DoTheBestToGetTheBest** [reply]
>I left some comments on your recent pr,

Thank you so so much for this

>do you have keybase,twitter? anything where we can DM, communicating via gh discussions has it's limitations hehe.

Yes i have twitter, thank you for asking this ! https://twitter.com/UnityGameStart

>I have something in mind which is a bit more advanced, but I think you can take it on.


Thank you so much for giving me this opportunity. I'am happy to take this one and do my best to solve it :)


>wdyt?

Thank you very much for the help you give me and the explanation and also for the access to the node. I'm really happy to take this issue !

**Comment by DoTheBestToGetTheBest** [comment]
> I left some comments on your recent pr, do you have keybase,twitter? anything where we can DM, communicating via gh discussions has it's limitations hehe.
> 
> I have something in mind which is a bit more advanced, but I think you can take it on.
> 
> we need more support for rpc testing. we have these extension helpers for example:
> 
> https://github.com/paradigmxyz/reth/blob/9dd270e4e8095a4cab1d2118f8f965eec68f4db7/crates/rpc/rpc-testing-util/src/trace.rs#L23-L37
> 
> but we also want more, like a stream for `replayTransaction` that takes a list of hashes and returns the traced results in a stream. this requires some additional knowledge about futures/streams etc, but I think you can take this now. happy to give access to a node so you can test this properly.
> 
> wdyt?

Hello Mat, hope you're doing good. Should i create issue for this ?

**Comment by DoTheBestToGetTheBest** [comment]
First i'm going to create a similar type to TraceBlockResult
```rust
pub type ReplayTransactionResult = Result<(LocalizedTransactionTrace, TxHash), (RpcError, TxHash)>;
```

and in traceApiExt i will define two function

```rust
fn replay_transaction_buffered
fn replay_transaction_buffered_unordered
``` 


and finally a struct 

```rust
pub struct ReplayTransactionStream<'a> {
    stream: Pin<Box<dyn Stream<Item = ReplayTransactionResult> + 'a>>,
}
``` 

which i will implement stream for my struct and debug

**Comment by DoTheBestToGetTheBest** [comment]
@mattsse hello boss, good week for you ! I'am ready to take new issue for this week :) ( but please this time i have lot homework and university stuff so i will not be able to work on it everytime, i hope you can understand that ) All my free time will be taken for reth. Thank you !

**Comment by DoTheBestToGetTheBest** [comment]
hey @mattsse hope you're doing well, anything i can take ? I know i have that issue which is not finished but would go on it next time. Wanna do something else if possible :)

I will return back to that issue when i'm more ready
